UIforETW/Utility.cpp

```diff
@@ -582,18 +805,18 @@ typedef struct tagTHREADNAME_INFO
 // warning C6320 : Exception - filter expression is the constant EXCEPTION_EXECUTE_HANDLER.This might mask exceptions that were not intended to be handled.
 // warning C6322 : Empty _except block.
 
-void SetCurrentThreadName(PCSTR threadName)
+void SetCurrentThreadName(PCSTR const threadName)
 {
-	const DWORD dwThreadID = GetCurrentThreadId();
-	THREADNAME_INFO info;
-	info.dwType = 0x1000;
-	info.szName = threadName;
-	info.dwThreadID = dwThreadID;
-	info.dwFlags = 0;
-
+	const DWORD dwThreadID = ::GetCurrentThreadId();
+	const THREADNAME_INFO info = { 0x1000, threadName, dwThreadID, 0 };
 	__try
 	{
-		RaiseException(MS_VC_EXCEPTION, 0, sizeof(info) / sizeof(ULONG_PTR), (ULONG_PTR*)&info);
+		const DWORD numArguments = sizeof(info) / sizeof(ULONG_PTR);
+		static_assert(

```

Comment:
It seems a bit excessive to test Microsoft's sample code for validity.

Remember, this is a tool, not a test framework.